File: internal/app/container_api.go
package app

import (
	"context"

	"nens-k8s/internal/domain"
)

type ContainerAPI struct {
	containers domain.ContainerResolver
	ctx        context.Context
}

func NewContainerAPI(containers domain.ContainerResolver) *ContainerAPI {
	return &ContainerAPI{containers: containers}
}

func (a *ContainerAPI) bind(ctx context.Context) {
	a.ctx = ctx
}

func (a *ContainerAPI) Targets(ref domain.ResourceRef) ([]domain.ContainerTarget, error) {
	return a.containers.Targets(a.ctx, ref)
}

File: frontend/src/features/containers/container.api.ts
import { Targets } from '@bindings/go/app/ContainerAPI'
import type { domain } from '@bindings/go/models'
import { useClusters } from '@/features/clusters/cluster.store'
import type { ResourceRef } from '@/features/resources/resource.types'
import { fixtureTargets } from './container.fixtures'
import type { ContainerTarget } from './container.types'

export async function listContainers(ref: ResourceRef): Promise<ContainerTarget[]> {
  if (useClusters.getState().offline) return fixtureTargets(ref)
  return (await Targets(ref as domain.ResourceRef)) as ContainerTarget[]
}
